client/cli/src/commands/vanity.rs

```diff
@@ -81,34 +98,62 @@ where
 {
 	println!("Generating key containing pattern '{}'", desired);
 
-	let top = 45 + (desired.len() * 48);
-	let mut best = 0;
-	let mut seed = Pair::Seed::default();
-	let mut done = 0;
-
-	loop {
-		if done % 100000 == 0 {
-			OsRng.fill_bytes(seed.as_mut());
+	let top = 45 + (desired.len() * 48) + {
+		if left {
+			1
 		} else {
-			next_seed(seed.as_mut());
+			0
 		}
+	};
 
-		let p = Pair::from_seed(&seed);
-		let ss58 = p.public().into_account().to_ss58check_with_version(network_override);
-		let score = calculate_score(&desired, &ss58);
-		if score > best || desired.len() < 2 {
-			best = score;
-			if best >= top {
-				println!("best: {} == top: {}", best, top);
-				return Ok(utils::format_seed::<Pair>(seed.clone()))
-			}
-		}
-		done += 1;
+	let (tx, rx) = channel();
 
-		if done % good_waypoint(done) == 0 {
-			println!("{} keys searched; best is {}/{} complete", done, best, top);
-		}
+	for core in 0..cores {
+		let network_override = network_override.clone();
+		let des = desired.clone().to_string();
+		let sender = tx.clone();
+		thread::spawn(move || {
+			let mut best = 0;
+			let mut seed = Pair::Seed::default();
+			let mut done = 0;
+
+			loop {
+				if done % 100000 == 0 {
+					OsRng.fill_bytes(seed.as_mut());
+				} else {
+					next_seed(seed.as_mut());
+				}
+
+				let p = Pair::from_seed(&seed);
+				let ss58 = p.public().into_account().to_ss58check_with_version(network_override);
+				let score = calculate_score(&des, &ss58);
+				if score > best || des.len() < 2 {
+					best = score;
+					if best >= top {
+						println!("best: {} == top: {}", best, top);
+						sender.send(utils::format_seed::<Pair>(seed.clone())).expect("Failed to send generated address");
```

Comment:
```suggestion
						let _ = sender.send(utils::format_seed::<Pair>(seed.clone()));
```

You should not panic here.